src/systems/joint_trajectory_controller/JointTrajectoryController.cc

```diff
@@ -0,0 +1,795 @@
+    ignition::math::PID velocity;
+  } pids;
+};
+
+/// \brief Information about trajectory that is followed by JointTrajectoryController plugin
+class Trajectory
+{
+public:
+  /// \brief Update index of trajectory points, such that it directs to a point that needs to be
+  /// currently followed
+  /// \param[in] _simTime Current simulation time
+  /// \return True if index of the trajectory point was updated, False otherwise
+  bool UpdateCurrentPoint(const std::chrono::steady_clock::duration &_simTime);
+
+  /// \brief Determine if the trajectory goal was reached
+  /// \return True if trajectory goal was reached, False otherwise
+  bool IsGoalReached() const;
+
+  /// \brief Compute progress of the current trajectory
+  /// \return Fraction of the completed points in range [0.0, 1.0]
+  float ComputeProgress() const;
+
+  /// \brief Reset trajectory internals, i.e. clean list of joint names, points and reset index
+  /// of the current point
+  void Reset();
+
+  /// \brief Status of the trajectory
+  enum TrajectoryStatus
+  {
+    /// \brief Trajectory is new and needs to be configure on the next update loop
+    New,
+    /// \brief Trajectory is currently being followed
+    Active,
+    /// \brief Trajectory goal is reached
+    Reached,
+  } status;
+
+  /// \brief Start time of trajectory
+  std::chrono::steady_clock::duration startTime;
+
+  /// \brief Index of the current trajectory point
+  unsigned int pointIndex;
+
+  /// \brief Ordered joints that need to be actuated to follow the current trajectory
+  std::vector<std::string> jointNames;
+
+  /// \brief Trajectory defined in terms of temporal points, whose members are ordered according
+  /// to `jointNames`
+  std::vector<ignition::msgs::JointTrajectoryPoint> points;
+};
+
+/// \brief Private data of the JointTrajectoryController plugin
+class ignition::gazebo::systems::JointTrajectoryControllerPrivate
+{
+public:
+  /// \brief Callback for joint trajectory subscription
+  /// \param[in] _msg A new message describing a joint trajectory that needs to be followed
+  void JointTrajectoryCallback(const ignition::msgs::JointTrajectory &_msg);
+
+  /// \brief Configure a single joint so that it can be actuated to follow a trajectory
+  /// \param[in] _entity Entity of the joint
+  /// \param[in] _sdf SDF reference used to obtain configuration for this joint
+  /// \param[in] _ecm Ignition Entity Component Manager
+  /// \param[in] _enabledJoints List of all joints that should be enabled based on SDF configuration
+  void ConfigureJoint(const Entity &_entity,
+                      const std::shared_ptr<const sdf::Element> &_sdf,
+                      const ignition::gazebo::EntityComponentManager &_ecm,
+                      const std::vector<std::string> &_enabledJoints);
+
+  /// \brief Reset internals of the plugin, without affecting already created components
+  void Reset();
+
+  /// \brief Ignition communication node
+  transport::Node node;
```

Comment:
I have just noticed using `public` keyword  in one of the other plugins for each member, e.g.:
```cpp
public: transport::Node node;
```

Should it be changed here?

Reply:
Yes, please :pray: Thanks!